### PR TITLE
safer check for isatty in fx/_utils.py

### PR DESCRIPTION
**Pull request by GoatGirl98**
if no isatty method is defined, it's probably not a tty

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

### CODE REVIEW

File: torch/fx/_utils.py [hunk 1]
```diff
@@ -20,8 +20,10 @@ def format_name():
     if "print_output" not in kwargs:
         kwargs["print_output"] = False
 
-    if "colored" in kwargs and not sys.stdout.isatty():
-        kwargs["colored"] = False
+    _stream_stdout = getattr(sys, 'stdout')
```
Comment (ezyang) [inline torch/fx/_utils.py:23]: Is there a reason to do this? It seems equivalent to accessing sys.stdout

**Review by ezyang — APPROVED** [review]

File: torch/fx/_utils.py (hunk 1 above)
Comment (GoatGirl98) [reply torch/fx/_utils.py:23]: > Is there a reason to do this? It seems equivalent to accessing sys.stdout

I changed `sys.stdout`  to an object that does not imitate `_io.TextIOWrapper` in my project, so there is no method `isatty` in this case and will raise AttributeError.

> torch._dynamo.exc.InternalTorchDynamoError: AttributeError: 'F' object has no attribute 'isatty'

Comment (ezyang) [reply torch/fx/_utils.py:23]: Specifically, why say `getattr(sys, 'stdout')` when `sys.stdout` seems like it would work

Comment (ezyang) [reply torch/fx/_utils.py:23]: yeah lint complaining about it

Comment (GoatGirl98) [reply torch/fx/_utils.py]: > yeah lint complaining about it

My bad, sorry for the inconvenience. Unnecessary code has been removed in the latest commit in this PR.

**Comment by GoatGirl98** [comment]
> @pytorchbot merge

I've made further changes trying to fix the issue for lint no-clang, could you please merge it again? Sorry for the inconvenience.  :)

**Comment by ezyang** [comment]
Meh FIIINE

**Pull request closed without merging**